openlibrary/solr/update_work.py

```diff
@@ -977,7 +977,7 @@ def toxml(self):
         node = dict2element(self.doc)
         root = Element("add")
         root.append(node)
-        return tostring(root).encode('utf-8')
+        return six.ensure_text(tostring(root))
```

Comment:
It seems like the previous version is just incorrect code that ends up being a NOOP in Python2 --
`return tostring(root).decode('utf-8')`  seems to be what was intended originally and that doesn't break in Python3

The XML content seems safely ASCII here. Double utf8 encoding can cause problems when non-ASCII characters are involved.

lmxl's `.tostring()` returns bytes according to this tutorial https://lxml.de/tutorial.html, so decoding seems to be what is required here. There's more details about `tostring()` here https://lxml.de/api/lxml.etree-module.html#tostring

This seems less about Python3 compatibility and more about Python2 masking bugs.